clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -71,13 +58,15 @@
     private final int writeLimit;
     private final int initialCapacity;
 
+    private volatile float estimatedCompressionRatio;
+
     private boolean appendStreamIsClosed = false;
     private long producerId;
     private short producerEpoch;
     private int baseSequence;
     private long writtenUncompressed = 0;
     private int numRecords = 0;
-    private float compressionRate = 1;
+    private float compressionRatio = 1;
```

Comment:
Consider renaming this to `actualCompressionRatio` just to enhance the distinction from `estimatedCompressionRatio`.